R/new_city.R

```diff
@@ -48,9 +51,16 @@
 #' }
 #' @export
 
-new_city <- function(name = NULL, country = NULL, lat = NULL, long = NULL) {
-  stopifnot("specify all input arguments" = all(c(!is.null(name), !is.null(country), !is.null(lat), !is.null(long))))
-  stopifnot("all input must be of length 1" = all(c(length(name) == 1, length(country) == 1, length(lat) == 1, length(long) == 1)))
+new_city <- function(name = NULL, country = NULL, lat = NULL, long = NULL, method = "osm") {
+  # Use match.arg to validate the method parameter
+  method <- match.arg(method, choices = c("osm", "census", "arcgis", "census_simple", "geocodio",
+                                          "mapbox", "google", "bing", "here", "tomtom", "nominatim", "tiger"))
+  stopifnot("At least provide a location name and country" = all(c(length(name) == 1, length(country) == 1)))
+  if(is.null(lat) || is.null(long)){
+    out = .geocode(name, country)
```

Comment:
Please use `<-` instead of `=`.

Reply:
Also run `styler::style_pkg()` in the R console when your working directory is set to the package folder to automatically adhere to the tidyverse style guide (https://style.tidyverse.org)